spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderSupportAutoConfiguration.java

```diff
@@ -60,29 +64,61 @@ public KafkaStreamsBinderConfigurationProperties binderConfigurationProperties(K
 	}
 
 	@Bean("streamConfigGlobalProperties")
-	public Map<String, Object> streamConfigGlobalProperties(KafkaStreamsBinderConfigurationProperties binderConfigurationProperties) {
-		Map<String, Object> props = new HashMap<>();
-		props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, binderConfigurationProperties.getKafkaConnectionString());
-		props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.ByteArraySerde.class.getName());
-		props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.ByteArraySerde.class.getName());
-		props.put(StreamsConfig.APPLICATION_ID_CONFIG, binderConfigurationProperties.getApplicationId());
+	public Map<String, Object> streamConfigGlobalProperties(KafkaStreamsBinderConfigurationProperties binderConfigurationProperties,
+															Environment environment) {
+
+		//Initialize configuration with default boot auto configuration properties
+		Map<String, Object> streamsConfiguration =
+				new HashMap<>(binderConfigurationProperties.getKafkaProperties().buildStreamsProperties());
```

Comment:
I think you should use the `KafkaStreamsConfiguration` `@Bean` to take advantage of boot's auto-configuring the `spring.application.name` to the `application.id` if not explicitly set.